pandas/tests/dtypes/test_common.py

```diff
@@ -706,3 +707,24 @@ def test__get_dtype_fails(input_param, expected_error_message):
 )
 def test__is_dtype_type(input_param, result):
     assert com._is_dtype_type(input_param, lambda tipo: tipo == result)
+
+
+@pytest.mark.parametrize("from_type", [np.datetime64, np.timedelta64])
+@pytest.mark.parametrize(
+    "to_type",
+    [
```

Comment:
can you add a test for the object path (for dtype=) when its a datetime/timedelta

Reply:
What would the expected output be for that case?  I would think it'd agree with `astype` from `numpy` when we don't have any NA values but the result is a bit odd:
```
[ins] In [3]: arr = np.array([np.datetime64("2018")])                                          

[ins] In [4]: arr.astype("object")                                                             
Out[4]: array([datetime.date(2018, 1, 1)], dtype=object)

[ins] In [5]: astype_nansafe(arr, dtype="object")                                              
Out[5]: array([datetime.datetime(1970, 1, 1, 0, 0)], dtype=object)
```